### ISSUE TITLE
Nuget push package to azure with authentication enabled

### ISSUE THREAD

**Issue by martincv8** [NuGet/Home#4548]
Hi Guys, 

I'm having a slight problem with pushing packages to a Nuget server on Azure when authentication is enabled, so yes, the push works absolutely fine before authentication is switched on.  

After enabled the authentication, I get taken to the MS sign in page when accessing the URL in my browser, when attempting to push a package I am presented with a message :  "Please provide credentials for: URL"

If I enter the exact same credentials, I am asked for credentials again with no further explanation of why.  When firing up Fiddler I can see that there is a GET request resulting in a 401.

I have specified the credentials to use using the nuget sources command (And even saved it in plainText using  -StorePasswordInClearText)

My command is currently formatted like this:

nuget push [nameOfPackage].nupkg -Source [URL]
Pushing [nameOfPackage].nupkg to '[URL]/api/v2/package'...
Using credentials from config. UserName: [MyUsername]
Please provide credentials for: [URL]/api/v2/package

In the meantime I have added a new HTTP Module to ask for authentication and I am specifying the username and password in the config.  This will work as a temporary measure however : 

1)  I feel a little silly that I haven't been able to figure this out.
2)  I get a very strong sense that I'm going down a rabbit hole and reinventing something that really shouldn't be necessary. 
3) When I started numbering this I was sure I had more points as to why what I am doing now is stupid.

I am also using an API key I saved against the source, if that matters?

This is the complete header being returned in Fiddler:

HTTP/1.1 401 Unauthorized
Content-Length: 58
Content-Type: text/html
Server: Microsoft-IIS/8.0
WWW-Authenticate: Bearer realm="[URL]"
X-Powered-By: ASP.NET
Set-Cookie: ARRAffinity=[shortened];Path=/;Domain=[URL]
Date: Thu, 09 Feb 2017 15:02:41 GMT
Connection: close

You do not have permission to view this directory or page.

**Comment by rrelyea** [comment]
Great issue... But missing one thing critical to understanding...  What versions of Nuget.exe have you tried? 
Please make sure you try 3.5 and  4.0.0-rc4.
Let us know how those all do.  Thx.

**Comment by martincv8** [comment]
@rrelyea Thanks for the quick response.  Admittedly, I was using an older version (and I don't feel stupid for not checking this before logging the bug, at all :)).  That being said I've issued a nuget update -self which pushed me to version 3.5.0 and I'm still seeing the exact same problem. 

I then got the RC and I am seeing the same behaviour, the one thing that I did see change is that I now get a little more feedback indicating a PUT request being used.  It still loads up credentials from config and even re-entering them myself I'm seeing the same behaviour.  For my own sanity I switched off the auth and confirmed (again) that I could upload and delete the package, on all 3 versions of nuget.

So in summary I have used these two as the latest version:
NuGet Version: 4.0.0.2275
NuGet Version: 3.5.0.1996

I am seeing a rather ominious looking Expect: 100-continue in my the fiddler request...

PUT /api/v2/package/ HTTP/1.1
X-NuGet-ApiKey: [MYKEY]
user-agent: NuGet Command Line/3.5.0 (Microsoft Windows NT 6.2.9200.0)
Accept-Language: en-US
Content-Type: multipart/form-data; boundary=[GUID]
Host: [HostName]
Expect: 100-continue
Accept-Encoding: gzip, deflate
Connection: Keep-Alive
Content-Length: 4700
